### ISSUE TITLE
It will be more convenient if an extra param such as 'key' be added to dataTransform

### ISSUE THREAD

**Issue by hustcer** [timoxley/columnify#36]
Hi,
Thanks for the great work, and I think It will be more convenient if an extra param such as 'key' be added to dataTransform, sometimes we need to transform data according to which key current data is.

**Comment by timoxley** [comment]
@hustcer good idea, you can get it from reading the name property of the second argument to the `dataTransform`. This isn't documented though, so my bad. If you get this working, think you could send a docs PR?

**Comment by hustcer** [comment]
I have tried and the second argument just displays nothing but `{}`, I got no key name? Is there anything wrong?

**Comment by kmgilbert100** [comment]
@hustcer The same was happening with me... I added a PR that adds a 4th param called columnName not sure if this should be closed if theres some other fixed needed with the second argument?

**Comment by kmgilbert100** [comment]
@timoxley Looks like in the data you have to specify a name key to populate that column object? If you look at the example in my pr its how I am attempting to accomplish.. hoping can get that merged soon so I can pull it in from master to my project .. got it setup via npm link working exactly how I need it.. very useful feature :)
